app/AppInit.jsx

```diff
@@ -54,7 +55,34 @@ class AppInit extends React.Component {
         };
     }
 
+    componentDidCatch(error, errorInfo) {
+        LogsActions.setLog({
+            type: "query",
+            log: {error, errorInfo}
+        });
+    }
+
     componentWillMount() {
+        const saveLog = (type, log) =>
+            LogsActions.setLog({type, log: Array.from(log)});
+
+        console.log = function() {
```

Comment:
The overwritten console function should still print to the console log while also filling up the new log storage